### ISSUE TITLE
Indentation check do not raise violations for incorrectly indented methods returning arrays

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#1430]
The following code should raise violation with any reasonable configuration for Indentation check:
```java
class A {
String[] incorrectlyIndentedMethodReturningArrayOfStrings() {return null;} // no indentation here
}
```
but it doesn't, due to incorrect implementation in https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodDefHandler.java#L55

**Comment by robertpainsi** [comment]
Hi! I opened the supplied example with the Checkstyle Graph GUI (and also did a quick debug session with the [current master](https://github.com/checkstyle/checkstyle/tree/a52c7a7a8ae8500a46bd998a1782e96dc481325d) with the same result).

![image](https://cloud.githubusercontent.com/assets/1794599/22646910/0c5d181c-ec6f-11e6-92e5-1cd0a09dbf1f.png)

As you can see, the `METHOD_DEF` doesn't cover `String`. Is this intended?

**Comment by rnveach** [comment]
@robertpainsi That looks like a different issue with the GUI and how the internal Tree isn't in the same order as the source code.
[Running the print tree option in the CLI on the code](http://checkstyle.sourceforge.net/cmdline.html#Command_line_usage), shows the String is indeed part of the `METHOD_DEF`:
````
    |--METHOD_DEF -> METHOD_DEF [2:6]
    |  |--MODIFIERS -> MODIFIERS [2:6]
    |  |--TYPE -> TYPE [2:6]
    |  |  `--ARRAY_DECLARATOR -> [ [2:6]
    |  |      |--IDENT -> String [2:0] <---------------------------- String is here
    |  |      `--RBRACK -> ] [2:7]
    |  |--IDENT -> incorrectlyIndentedMethodReturningArrayOfStrings [2:9]
    |  |--LPAREN -> ( [2:57]
    |  |--PARAMETERS -> PARAMETERS [2:58]
    |  |--RPAREN -> ) [2:58]
    |  `--SLIST -> { [2:60]
````

The issue with indentation here is the `[]` is out of order compare to the `String`. We see `[` first and not `String`, so we are probably thinking the `[` is first on the line. If you remove the `[]` you will get a violation from Checkstyle.
`IndenationCheck` works best when the tree is in the same order as the source, otherwise we get these weird bugs. IMO, we should be creating the tree in source order but we should achieve that if we ever move to antlr4, where it is forced by default.
